lib/rules/declaration-property-max-values/index.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const matchesStringOrRegExp = require('../../utils/matchesStringOrRegExp');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const valueParser = require('postcss-value-parser');
+const vendor = require('../../utils/vendor');
+const validateOptions = require('../../utils/validateOptions');
+const { isNumber } = require('../../utils/validateTypes');
+const validateObjectWithProps = require('../../utils/validateObjectWithProps');
+
+const ruleName = 'declaration-property-max-values';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (property, max) =>
+		`Expected "${property}" to have no more than ${max} ${max === 1 ? 'value' : 'values'}`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/declaration-property-max-values',
+};
+
+const isValueNode = (/** @type {valueParser.Node} */ node) => {
```

Comment:
[trivial] The following may be more consistent with other code:

```suggestion
/**
 * @param {valueParser.Node} node
 */
const isValueNode = (node) => {
```